inference-engine/src/gna_plugin/gna_plugin.hpp

```diff
@@ -115,7 +115,7 @@ class GNAPlugin : public InferenceEngine::IInferencePluginInternal, public std::
                       const std::map<std::string, std::string>& config,
                       InferenceEngine::QueryNetworkResult &res) const override;
     uint32_t QueueInference(const InferenceEngine::BlobMap &input, InferenceEngine::BlobMap &result);
-    bool Wait(uint32_t idx = 0);
+    bool Wait(uint32_t idx, int64_t millis_timeout = -1);
```

Comment:
i would use overload with and without timeout. Or create new method waitFor()  - this will encapsulate magic number in single default value specified in gna_device helper.

Reply:
We have access here to MAX_TIMEOUT, so I decided it's easier just to reuse it here. I also made the parameter camel case, for consistency with the rest of the plugin code.

Reply:
That solution was incorrect. Replaced with the one you proposed.